src/sentry/quotas/redis.py

```diff
@@ -192,37 +228,48 @@ def is_rate_limited(self, project, key=None, timestamp=None):
         if timestamp is None:
             timestamp = time()
 
-        quotas = self.get_quotas_with_limits(project, key=key)
+        quotas = self.get_quotas(project, key=key)
 
         # If there are no quotas to actually check, skip the trip to the database.
         if not quotas:
             return NotRateLimited()
 
+        rejections = []
         keys = []
         args = []
         for quota in quotas:
+            if quota.limit == 0:
```

Comment:
It might be nice to also check `.is_tracked()` here. There are only four cases at the moment:
 1. Untracked zero quota: Reject immediately. We can assert that `limit==0`.
 2. Tracked limited quota: Send to redis
 3. Tracked unlimited quota: Send to redis.
 4. No quota: This code path will never be hit.